sdk/python/v1beta1/kubeflow/katib/api/katib_client.py

```diff
@@ -329,6 +329,52 @@ def list_trials(self, name=None, namespace=None):
           {2} ".format(name, namespace, e))
         return result
 
+    def get_success_trial_details(self, name=None, namespace=None):
```

Comment:
Maybe we should have an API which returns all Experiment's Trials objects instead of only Succeeded Trials with name, hyperparams, metrics?

Then user will have more freedom with this list of Trials ?
E.g. Get all Succeeded Trials, Get Early Stopped Trials, Get top <k> Trials.

WDYT @Adarsh2910 @gaocegege ?

Reply:
SGTM, should we add it in this PR? or open an issue for it?

Reply:
Do you think it's a good idea to give the user a way to filter the trials based on status, or just return the list of trials with hyperparams and metrics?

If it's the latter, we can also modify the `list_trials` method and just add hyperparams and metrics to the return.

Reply:
From my point of view, `list_trials` should return the list of `Trial` objects, instead of only name and status.
We can also add function: `get_success_trial_details` if it is needed in addition to `list_trials`.
What do you think @Adarsh2910 @gaocegege  ?

Reply:
That sounds good to me
I think having a function to just get the success trials will be a helpful addition and we can raise another issue for patching `list_trials` to get the list of `Trial` objects

Reply:
If we are all in agreement, Can we go ahead and merge this one in?
Thoughts? @andreyvelich @gaocegege

Reply:
Make sense, we can update `list_trials` in the next PR.
WDYT @gaocegege ?

Reply:
SGTM.